File: shared/contracts/enums.ts
export const ROOM_STATUSES = [
  "waiting",
  "starting",
  "in_game",
  "finished",
] as const;

export type RoomStatus = (typeof ROOM_STATUSES)[number];

export const GAME_PHASES = [
  "lobby",
  "setup",
  "turn_action",
  "turn_draw",
  "resolving_explosion",
  "finished",
] as const;

export type GamePhase = (typeof GAME_PHASES)[number];

export const CARD_TYPES = [
  "exploding_kitten",
  "defuse",
  "skip",
  "attack",
  "shuffle",
  "see_the_future",
  "favor",
] as const;

export type CardType = (typeof CARD_TYPES)[number];

export const PLAYER_STATUSES = [
  "connected",
  "disconnected",
  "eliminated",
] as const;

export type PlayerStatus = (typeof PLAYER_STATUSES)[number];

export const ACTION_TYPES = [
  "start_game",
  "play_skip",
  "play_attack",
  "play_shuffle",
  "play_see_the_future",
  "play_favor",
  "draw_card",
  "defuse",
  "explode",
  "eliminate",
  "turn_advanced",
] as const;

export type ActionType = (typeof ACTION_TYPES)[number];

File: backend/app/schemas/enums.py
from enum import StrEnum


class RoomStatus(StrEnum):
    WAITING = "waiting"
    STARTING = "starting"
    IN_GAME = "in_game"
    FINISHED = "finished"


class GamePhase(StrEnum):
    LOBBY = "lobby"
    SETUP = "setup"
    TURN_ACTION = "turn_action"
    TURN_DRAW = "turn_draw"
    RESOLVING_EXPLOSION = "resolving_explosion"
    FINISHED = "finished"


class CardType(StrEnum):
    EXPLODING_KITTEN = "exploding_kitten"
    DEFUSE = "defuse"
    SKIP = "skip"
    ATTACK = "attack"
    SHUFFLE = "shuffle"
    SEE_THE_FUTURE = "see_the_future"
    FAVOR = "favor"


class PlayerStatus(StrEnum):
    CONNECTED = "connected"
    DISCONNECTED = "disconnected"
    ELIMINATED = "eliminated"


class ActionType(StrEnum):
    START_GAME = "start_game"
    PLAY_SKIP = "play_skip"
    PLAY_ATTACK = "play_attack"
    PLAY_SHUFFLE = "play_shuffle"
    PLAY_SEE_THE_FUTURE = "play_see_the_future"
    PLAY_FAVOR = "play_favor"
    DRAW_CARD = "draw_card"
    DEFUSE = "defuse"
    EXPLODE = "explode"
    ELIMINATE = "eliminate"
    TURN_ADVANCED = "turn_advanced"
